### DISCUSSION TITLE
How to implement a search with Elasticlunr for example?

### DISCUSSION THREAD

**Discussion by matthieu2607** [Cecilapp/Cecil#1467] [Q&A] [answered]
Thank you Arnaud for your great work ;)

I would like to feed an index used by Elasticlunr when I compile the site to implement a search on the site.

I would like to be able to add the title and the tags in this index for all the pages created.

I looked in the documentation but couldn't find anything on this point.

Is it possible ?

Thank you in advance for your return.

Best regards,

Matthieu

**Comment by ArnaudLigny** [comment, marked as answer]
Hello @matthieu2607, I thinks the best to do that is using the [output/formats feature](https://cecil.app/documentation/configuration/#formats):
1. create a new format dedicated to elasticlunr
2. associate this format to homepage type (to list all pages)
3. create a layout to format the JSON

Example:
- output/format: https://github.com/ArnaudLigny/arnaudligny.fr/blob/master/config.yml#L52
- layout: https://github.com/ArnaudLigny/arnaudligny.fr/blob/master/layouts/_default/list.elasticlunr.twig
- result: https://arnaudligny.fr/elasticlunr.json

**Comment by matthieu2607** [comment]
Hi @ArnaudLigny,

Big thanks for your reply ;)

I'm going to use this and I'm finally thinking of going to fuse.js

Good day
